br/pkg/storage/s3.go

```diff
@@ -591,8 +591,38 @@ func (rs *S3Storage) WriteFile(ctx context.Context, file string, data []byte) er
 	return errors.Trace(err)
 }
 
-// ReadFile reads the file from the storage and returns the contents.
 func (rs *S3Storage) ReadFile(ctx context.Context, file string) ([]byte, error) {
+	backoff := 10 * time.Millisecond
+	remainRetry := 5
+	contRetry := func() bool {
+		if remainRetry <= 0 {
+			return false
+		}
+		time.Sleep(backoff)
+		remainRetry -= 1
+		return true
+	}
+
+	// The errors cannot be handled by the SDK because they happens during reading the HTTP response body.
+	// We cannot use `utils.WithRetry[V2]` here because cyclinic deps.
+	for {
+		data, err := rs.doReadFile(ctx, file)
+		if err != nil {
+			log.Warn("ReadFile: failed to read file.",
```

Comment:
nit, feels like it makes more sense to log after we exhausted the remaining retry.